src/blas/dasum.c

```diff
@@ -20,19 +20,19 @@ doublereal dasum_(integer *n, doublereal *dx, integer *incx)
     doublereal ret_val, d__1, d__2, d__3, d__4, d__5, d__6;
 
     /* Local variables */
-    static integer i, m;
-    static doublereal dtemp;
-    static integer nincx, mp1;
+    integer i, m;
```

Comment:
Why did you remove all of the `static` keywords?  We generally try to change the blas code as minimally as possible.  We don't even apply astyle indentation to the files.

Reply:
Hi Rob, this change was needed so that I'm able to call blas/lapack routines inside `omp parallel` regions. There are race conditions If these variable are `static` because they are initialized only once and shared between all threads executing that code. We can talk more if necessary :)

Reply:
Why blas/lapack wanted to use `static` for local variables? It doesn't make sense to me. P.S. I understood this code was translated by F2C from Fortran. Is this because of Fortran?

Reply:
I think `static` was used as an attempt to produce more optimized code, actually.  However, there will be an issue in openmp regions as Victor says.  Some of the early versions of this code were created via f2c, but I'm not sure about the CLAPACK release most recently used (see the `blas/README`).  Jack's group generally works hard at producing highly optimized code, so I wonder if it would be worthwhile to do a little more research about what's available for C implementations of blas, issues on thread safety, etc.  Maybe Victor has already done some of that?

Reply:
So, are we using a non thread-safe version of CLAPACK? Shall we upgrade our local blas/lapack? I would like to use the latest version if it's worth the effort of changing these files. Thanks!

Reply:
Hi all, yes, I did some research on this. It seems that support for CLAPACK has been discontinued (the last version is from 2009). The reference LAPACK implementation has been including a set of C interfaces (LAPACKE) in recent releases, see [this](https://github.com/Reference-LAPACK/lapack#lapacke). To my knowledge, pure C implementations are not being actively supported. I remember seeing in a recent CHANGELOG of lapack that some thread safety issues have been fixed (I'll try to find that). So, it would be nice to upgrade our own distribution; however, we would need to compile Fortran code.

Reply:
I don't want to require a Fortran compile for our provided version of blas/lapack.  Users can always link to external optimized versions of these routines.  The purpose of our provided code is to keep the compilation of hypre as simple as possible.  If there is a better C version of these routines that is thread safe, etc., then I am happy to switch to that.  I am also okay with keeping what is in this PR, but we should update the `README` file to mention removing `static` from new routines that get added in the future.

Reply:
I agree with you, Rob. Ruipeng, which functions are you planning to use? After the changes in this PR, I'm sure that `dpotrf` and `dpotrs` are thread-safe. Probably the more general dense direct solvers are thread-safe as well, but it would be good to double-check :).